src/Storages/ObjectStorage/StorageObjectStorageStableTaskDistributor.cpp

```diff
@@ -3,15 +3,21 @@
 #include <consistent_hashing.h>
 #include <optional>
 
+#include <Poco/JSON/Object.h>
+#include <Poco/JSON/Parser.h>
+#include <Poco/JSON/JSONException.h>
+
 namespace DB
 {
 
 StorageObjectStorageStableTaskDistributor::StorageObjectStorageStableTaskDistributor(
     std::shared_ptr<IObjectIterator> iterator_,
-    std::vector<std::string> ids_of_nodes_)
+    std::vector<std::string> ids_of_nodes_,
+    uint64_t lock_object_storage_task_distribution_ms_)
     : iterator(std::move(iterator_))
     , connection_to_files(ids_of_nodes_.size())
     , ids_of_nodes(ids_of_nodes_)
+    , lock_object_storage_task_distribution_us(lock_object_storage_task_distribution_ms_ * 1000)
```

Comment:
Poco::Timestamp is in microseconds, so here convert from milliseconds to microseconds to avoid multiplications or divides later.